### ISSUE TITLE
overlay is reverted when stock mount remount failed

### ISSUE THREAD

**Issue by 5ec1cff** [tiann/KernelSU#365]
最近两天有不少用户反馈 GSI 系统上 ksu 的模块功能不正常，看起来是因为[这个提交](https://github.com/tiann/KernelSU/commit/b76d973f3af4b01a33c7f852599410fd530003a8)导致。

相关代码：

https://github.com/tiann/KernelSU/blob/main/userspace/ksud/src/event.rs#L40

相关日志：

```
04-09 11:10:23.597   800   800 I KernelSU: ksud::mount: remount stock for /system/ : [MountInfo { source: "none", dest: "/system/system_ext/apex/com.android.vndk.v27/lib/libsoftkeymasterdevice.so", fstype: "tmpfs", options: ["rw", "seclabel", "nodev", "relatime", "mode=755"], dump: 0, pass: 0 }, MountInfo { source: "none", dest: "/system/system_ext/apex/com.android.vndk.v27/lib64/libsoftkeymasterdevice.so", fstype: "tmpfs", options: ["rw", "seclabel", "nodev", "relatime", "mode=755"], dump: 0, pass: 0 }, MountInfo { source: "none", dest: "/system/usr/keylayout", fstype: "tmpfs", options: ["rw", "seclabel", "nodev", "relatime", "mode=755"], dump: 0, pass: 0 }]
04-09 11:10:23.597   800   800 W KernelSU: ksud::event: remount stock failed: No such file or directory (os error 2), umount overlay /system
```

mounts：

```
none /system/system_ext/apex/com.android.vndk.v27/lib/libsoftkeymasterdevice.so tmpfs rw,seclabel,nodev,relatime,size=1877868k,nr_inodes=469467,mode=755 0 0
none /system/system_ext/apex/com.android.vndk.v27/lib64/libsoftkeymasterdevice.so tmpfs rw,seclabel,nodev,relatime,size=1877868k,nr_inodes=469467,mode=755 0 0
none /system/usr/keylayout tmpfs rw,seclabel,nodev,relatime,size=1877868k,nr_inodes=469467,mode=755 0 0
```

分析：/system 上存在 bind mount / tmpfs，ksud 无法正确 remount 它们。

相关链接：
https://github.com/tiann/KernelSU/issues/364
https://t.me/KernelSU_group/3252/62285
https://t.me/KernelSU_group/3252/62505
https://github.com/tiann/KernelSU/issues/233

**Comment by tiann** [comment]
https://t.me/KernelSU_group/27237/63464

**Comment by tiann** [comment]
should be fixed in: https://github.com/tiann/KernelSU/commit/a46d4ecd3ed0f8471f7f9667dd092e4b8b5d4f3b

we move the mounts now, maybe we can use overlayfs mount in the future.

**Comment by 5ec1cff** [comment]
> should be fixed in: [a46d4ec](https://github.com/tiann/KernelSU/commit/a46d4ecd3ed0f8471f7f9667dd092e4b8b5d4f3b)
> 
> we move the mounts now, maybe we can use overlayfs mount in the future.

A Rust version of my PoC has been completed: https://github.com/5ec1cff/ksu-mount-POC/tree/rust

**Comment by tiann** [comment]
PRs welcome!
